Index: app/src/main/java/com/shehabsalah/geranyapp/activities/MainActivity.java
===================================================================
--- app/src/main/java/com/shehabsalah/geranyapp/activities/MainActivity.java	(revision ab15501144c4075fcc31f40c938c983632c6f1c6)
+++ app/src/main/java/com/shehabsalah/geranyapp/activities/MainActivity.java	(revision ab15501144c4075fcc31f40c938c983632c6f1c6)
@@ -1,153 +0,0 @@
-package com.shehabsalah.geranyapp.activities;
-
-import android.content.Intent;
-import android.support.annotation.NonNull;
-import android.support.design.widget.BottomNavigationView;
-import android.os.Bundle;
-import android.util.Log;
-import android.view.MenuItem;
-import android.view.View;
-import android.widget.TextView;
-
-import com.firebase.ui.auth.AuthUI;
-import com.firebase.ui.auth.IdpResponse;
-import com.google.android.gms.tasks.OnCompleteListener;
-import com.google.android.gms.tasks.Task;
-import com.google.firebase.auth.FirebaseAuth;
-import com.google.firebase.auth.UserInfo;
-import com.shehabsalah.geranyapp.R;
-import com.shehabsalah.geranyapp.fragments.AddNewLocationFragment;
-import com.shehabsalah.geranyapp.main.ApplicationMain;
-import com.shehabsalah.geranyapp.util.Config;
-
-public class MainActivity extends ApplicationMain {
-    private static final String LOG_TAG = MainActivity.class.getSimpleName();
-    private UserInfo userInfo;
-
-    private TextView mTextMessage;
-    private BottomNavigationView.OnNavigationItemSelectedListener mOnNavigationItemSelectedListener
-            = new BottomNavigationView.OnNavigationItemSelectedListener() {
-
-        @Override
-        public boolean onNavigationItemSelected(@NonNull MenuItem item) {
-            switch (item.getItemId()) {
-                case R.id.navigation_home:
-                    mTextMessage.setText(R.string.title_home);
-                    return true;
-                case R.id.navigation_places:
-                    //ToDo: make the layout of places NOTE(must be recycler view) (#4)
-                    //ToDo: make the layout of places items (#5)
-                    //ToDo: make a fragment to view the places (#6)
-                    //ToDo: display recycler view with 10 fake items (#7)
-                    mTextMessage.setText(R.string.title_places);
-                    return true;
-                case R.id.navigation_post:
-                    //ToDo: make the layout of the new post (#8)
-                    //ToDo: make a fragment to display the view (#9)
-                    //ToDo: associate the fragment with the view and display it (#10)
-                    mTextMessage.setText(R.string.title_post);
-                    return true;
-                case R.id.navigation_add_place:
-                    /******************************Template****************************************/
-                    AddNewLocationFragment fragment = new AddNewLocationFragment();
-                    getSupportFragmentManager().beginTransaction()
-                            .add(R.id.home_container, fragment, Config.ADD_NEW_PLACE_FRAGMENT)
-                            .commit();
-                    /******************************Template****************************************/
-                    mTextMessage.setText(R.string.title_add_place);
-                    return true;
-                case R.id.navigation_profile:
-                    mTextMessage.setText(R.string.title_profile);
-                    return true;
-            }
-            return false;
-        }
-
-    };
-
-    @Override
-    protected void onCreate(Bundle savedInstanceState) {
-        super.onCreate(savedInstanceState);
-        setContentView(R.layout.activity_main);
-        mAuth = FirebaseAuth.getInstance();
-
-        //Initialize the FirebaseAuth instance and the AuthStateListener method so you can track
-        //whenever the user signs in or out.
-        mAuth = FirebaseAuth.getInstance();
-        if(isNotLoggedIn()){
-            signIn();
-        }
-        setFirebaseAuthListener();
-
-        userInfo = getUserInfo();
-        if (userInfo!=null){
-            Log.i(LOG_TAG, userInfo.getDisplayName());
-            Log.i(LOG_TAG, userInfo.getProviderId());
-            Log.i(LOG_TAG, userInfo.getEmail());
-            Log.i(LOG_TAG, userInfo.getUid());
-        }
-
-
-        /** Test Navigation */
-        mTextMessage = (TextView) findViewById(R.id.message);
-        BottomNavigationView navigation = (BottomNavigationView) findViewById(R.id.navigation);
-        navigation.setOnNavigationItemSelectedListener(mOnNavigationItemSelectedListener);
-        /** End Test Navigation **/
-
-    }
-
-
-
-    public void onClick(View v) {
-        //ToDo remove this method after adding it in the profile options (Settings)
-        if (v.getId() == R.id.sign_out) {
-            AuthUI.getInstance()
-                    .signOut(this)
-                    .addOnCompleteListener(new OnCompleteListener<Void>() {
-                        public void onComplete(@NonNull Task<Void> task) {
-                            /*
-                            User is now signed out, so restart the activity to apply the
-                             authentication layout.
-                             */
-                            Intent intent = getIntent();
-                            finish();
-                            startActivity(intent);
-                        }
-                    });
-        }
-    }
-
-    @Override
-    public void onBackPressed() {
-        finish();
-    }
-
-    @Override
-    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
-        super.onActivityResult(requestCode, resultCode, data);
-        // RC_SIGN_IN is the request code you passed into startActivityForResult(...) when starting the sign in flow.
-        if (requestCode == RC_SIGN_IN) {
-            IdpResponse response = IdpResponse.fromResultIntent(data);
-            // Handle SignIn States
-            handleSignInStates(resultCode,response);
-
-        }
-    }
-
-    @Override
-    public void onStart() {
-        super.onStart();
-        mAuth.addAuthStateListener(mAuthListener);
-    }
-
-    @Override
-    public void onStop() {
-        super.onStop();
-        if (mAuthListener != null) {
-            mAuth.removeAuthStateListener(mAuthListener);
-        }
-    }
-
-
-
-}
Index: app/src/main/java/com/shehabsalah/geranyapp/fragments/AddNewLocationFragment.java
===================================================================
--- app/src/main/java/com/shehabsalah/geranyapp/fragments/AddNewLocationFragment.java	(revision ab15501144c4075fcc31f40c938c983632c6f1c6)
+++ app/src/main/java/com/shehabsalah/geranyapp/fragments/AddNewLocationFragment.java	(revision ab15501144c4075fcc31f40c938c983632c6f1c6)
@@ -1,86 +0,0 @@
-package com.shehabsalah.geranyapp.fragments;
-
-import android.os.Bundle;
-import android.support.annotation.Nullable;
-import android.support.v4.app.Fragment;
-import android.support.v4.content.ContextCompat;
-import android.view.LayoutInflater;
-import android.view.View;
-import android.view.ViewGroup;
-import android.widget.Button;
-import android.widget.ImageView;
-import android.widget.TextView;
-
-import com.shehabsalah.geranyapp.R;
-import com.shehabsalah.geranyapp.util.Config;
-
-import butterknife.BindView;
-import butterknife.ButterKnife;
-
-/**
- * Created by ShehabSalah on 7/31/17.
- * This fragment will display when the user hit on (Add New Place) icon in Bottom navigation menu
- */
-
-public class AddNewLocationFragment extends Fragment {
-
-    @BindView(R.id.add_new_location_icon)
-    ImageView addNewLocationIcon;
-    @BindView(R.id.add_location_text)
-    TextView addLocationText;
-    @BindView(R.id.add_to_places_button)
-    Button addToPlacesButton;
-
-    @Nullable
-    @Override
-    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
-        View mainView = inflater.inflate(R.layout.add_new_location_layout, container, false);
-        ButterKnife.bind(this, mainView);
-        displayViews();
-        // ToDo: after the user click on add to my places button make the fullscreen black transparent layout that show square white box contain message (Do you like to give a name to your place?) and green right circle and gray close circle (#1.2)
-        // ToDo: if the user choose close: make the icon smooth changing from drop icon to beenhere icon and the text change to (Your current place is in your places list) [victor transition] (#1.3)
-        // ToDo: if the user choose yes: make the dialog change to add Place Name with input box, red cancel button and done blue button (#1.4)
-        return mainView;
-    }
-
-
-    /**
-     * This method responsible on display different views according to the internet connection and
-     * user current area. If there is no internet connection the method will hide addToPlacesButton
-     * button and display the no signal icon and message. If the device connected with the internet,
-     * the method will check the user location and display if the location exist in My Places list or
-     * not.
-     * */
-    private void displayViews(){
-        //Checking on the internet connection
-        if (Config.isNetworkConnected(getActivity())){
-            // if connected, then check if this place in My Places list or not
-            if (checkIfMyCurrentPlaceInPlacesListOrNot()) {
-                //This place is exist in My Places list.
-                addNewLocationIcon.setImageDrawable(ContextCompat.getDrawable(getActivity(), R.drawable.ic_beenhere));
-                addLocationText.setText(R.string.current_place_exist);
-                addToPlacesButton.setVisibility(View.GONE);
-            }else{
-                //This place not exist in My Places list.
-                //Enable Add New Place functionality.
-                addNewLocationIcon.setImageDrawable(ContextCompat.getDrawable(getActivity(), R.drawable.ic_person_pin_circle));
-                addLocationText.setText(R.string.add_new_place_text);
-                addToPlacesButton.setVisibility(View.VISIBLE);
-            }
-        }else{
-            // if not connected
-            addNewLocationIcon.setImageDrawable(ContextCompat.getDrawable(getActivity(), R.drawable.ic_signal_wifi_off));
-            addLocationText.setText(R.string.no_signal_message);
-            addToPlacesButton.setVisibility(View.GONE);
-        }
-    }
-    /**
-     * This Method responsible on comparing the user current place with his places list.
-     * @return (True) if the place exist and (False) if the place not exist
-     * */
-    private boolean checkIfMyCurrentPlaceInPlacesListOrNot(){
-        //ToDo: get the user current place and check if it exist in My Places list or not.
-        return false;
-    }
-
-}
Index: app/src/main/res/values/colors.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/res/values/colors.xml	(revision ab15501144c4075fcc31f40c938c983632c6f1c6)
+++ app/src/main/res/values/colors.xml	(revision )
@@ -12,5 +12,10 @@
     <color name="strokeColor">#6e404040</color>
 
 
-    <color name="white_text_color">#fff</color>
+    <color name="white_color">#fff</color>
+    <color name="black_color">#000</color>
+    <color name="black_light_color">#46000000</color>
+
+    <color name="default_icon_color">#8a000000</color>
+    <color name="success_color">#91dc5a</color>
 </resources>
Index: .idea/vcs.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- .idea/vcs.xml	(revision ab15501144c4075fcc31f40c938c983632c6f1c6)
+++ .idea/vcs.xml	(revision )
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project version="4">
   <component name="VcsDirectoryMappings">
-    <mapping directory="$PROJECT_DIR$" vcs="Git" />
+    <mapping directory="" vcs="Git" />
   </component>
 </project>
\ No newline at end of file
Index: app/build.gradle
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/build.gradle	(revision ab15501144c4075fcc31f40c938c983632c6f1c6)
+++ app/build.gradle	(revision )
@@ -5,7 +5,7 @@
     buildToolsVersion "26.0.0"
     defaultConfig {
         applicationId "com.shehabsalah.geranyapp"
-        minSdkVersion 19
+        minSdkVersion 21
         targetSdkVersion 25
         versionCode 1
         versionName "1.0"
@@ -35,6 +35,7 @@
     compile 'com.android.support:support-vector-drawable:25.3.1'
     compile 'com.jakewharton:butterknife:8.6.0'
     annotationProcessor 'com.jakewharton:butterknife-compiler:8.6.0'
+    compile 'com.android.support:recyclerview-v7:25.3.1'
     compile 'com.squareup.picasso:picasso:2.5.2'
     testCompile 'junit:junit:4.12'
 }
\ No newline at end of file
Index: app/src/main/res/layout/add_new_location_layout.xml
===================================================================
--- app/src/main/res/layout/add_new_location_layout.xml	(revision ab15501144c4075fcc31f40c938c983632c6f1c6)
+++ app/src/main/res/layout/add_new_location_layout.xml	(revision ab15501144c4075fcc31f40c938c983632c6f1c6)
@@ -1,53 +0,0 @@
-<?xml version="1.0" encoding="utf-8"?>
-<android.support.constraint.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
-    xmlns:app="http://schemas.android.com/apk/res-auto"
-    xmlns:tools="http://schemas.android.com/tools"
-    android:layout_width="match_parent"
-    android:layout_height="match_parent"
-    android:background="@color/appBackground"
-    tools:context=".fragments.AddNewLocationFragment">
-    <ImageView
-        android:id="@+id/add_new_location_icon"
-        android:layout_width="@dimen/ground_icon_width_size"
-        android:layout_height="@dimen/ground_icon_height_size"
-        app:srcCompat="@drawable/ic_person_pin_circle"
-        android:layout_marginRight="8dp"
-        app:layout_constraintRight_toRightOf="parent"
-        android:layout_marginLeft="8dp"
-        app:layout_constraintLeft_toLeftOf="parent"
-        android:layout_marginBottom="@dimen/shape_edge_padding"
-        app:layout_constraintBottom_toTopOf="@+id/add_location_text" />
-    <TextView
-        android:id="@+id/add_location_text"
-        android:layout_width="wrap_content"
-        android:layout_height="wrap_content"
-        android:textColor="@color/darkGray"
-        android:textSize="@dimen/text_size_large"
-        android:paddingStart="@dimen/shape_edge_padding"
-        android:paddingEnd="@dimen/shape_edge_padding"
-        android:text="@string/add_new_place_text"
-        android:textAlignment="center"
-        app:layout_constraintBottom_toBottomOf="parent"
-        android:layout_marginBottom="8dp"
-        android:layout_marginRight="8dp"
-        app:layout_constraintRight_toRightOf="parent"
-        android:layout_marginLeft="8dp"
-        app:layout_constraintLeft_toLeftOf="parent"
-        app:layout_constraintHorizontal_bias="0.375"
-        app:layout_constraintTop_toTopOf="parent"
-        android:layout_marginTop="8dp" />
-    <Button
-        android:id="@+id/add_to_places_button"
-        android:layout_width="wrap_content"
-        android:layout_height="wrap_content"
-        android:background="@drawable/button_bg"
-        android:text="@string/add_new_place_button"
-        android:textColor="@color/white_text_color"
-        android:layout_marginTop="@dimen/shape_edge_padding"
-        app:layout_constraintTop_toBottomOf="@+id/add_location_text"
-        android:layout_marginRight="8dp"
-        app:layout_constraintRight_toRightOf="parent"
-        android:layout_marginLeft="8dp"
-        app:layout_constraintLeft_toLeftOf="parent"
-        app:layout_constraintHorizontal_bias="0.502" />
-</android.support.constraint.ConstraintLayout>
\ No newline at end of file
Index: app/src/main/res/values/dimens.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/res/values/dimens.xml	(revision ab15501144c4075fcc31f40c938c983632c6f1c6)
+++ app/src/main/res/values/dimens.xml	(revision )
@@ -2,10 +2,28 @@
     <!-- Default screen margins, per the Android Design guidelines. -->
     <dimen name="activity_horizontal_margin">16dp</dimen>
     <dimen name="activity_vertical_margin">16dp</dimen>
+    <dimen name="extra_small_margin">5dp</dimen>
+    <dimen name="small_margin">11dp</dimen>
+    <dimen name="dialog_header_height">56dp</dimen>
     <dimen name="shape_edge_padding">16dp</dimen>
+    <dimen name="shape_edge_medium_padding">24dp</dimen>
+    <dimen name="shape_edge_medium_padding_minus">20dp</dimen>
+    <dimen name="shape_edge_extra_small_padding">8dp</dimen>
+    <dimen name="shape_edge_x2">32dp</dimen>
+    <dimen name="shape_edge_x4">64dp</dimen>
 
     <dimen name="ground_icon_width_size">80dp</dimen>
     <dimen name="ground_icon_height_size">80dp</dimen>
+    <dimen name="medium_icon_size">60dp</dimen>
+    <dimen name="icon_size">24dp</dimen>
 
-    <dimen name="text_size_large">20sp</dimen>
+    <dimen name="text_large_size">20sp</dimen>
+    <dimen name="text_normal_size_2">18sp</dimen>
+    <dimen name="text_medium_size">16sp</dimen>
+
+    <dimen name="text_normal_size">14sp</dimen>
+    <dimen name="text_extra_small_size">12sp</dimen>
+
+
+    <dimen name="bottom_actionbar_size">52dp</dimen>
 </resources>
Index: app/src/main/java/com/shehabsalah/geranyapp/main/ApplicationMain.java
===================================================================
--- app/src/main/java/com/shehabsalah/geranyapp/main/ApplicationMain.java	(revision ab15501144c4075fcc31f40c938c983632c6f1c6)
+++ app/src/main/java/com/shehabsalah/geranyapp/main/ApplicationMain.java	(revision ab15501144c4075fcc31f40c938c983632c6f1c6)
@@ -1,144 +0,0 @@
-package com.shehabsalah.geranyapp.main;
-
-import android.net.Uri;
-import android.support.annotation.NonNull;
-import android.support.v7.app.AppCompatActivity;
-import android.util.Log;
-
-import com.firebase.ui.auth.ErrorCodes;
-import com.firebase.ui.auth.IdpResponse;
-import com.firebase.ui.auth.ResultCodes;
-import com.google.firebase.auth.FirebaseAuth;
-import com.google.firebase.auth.FirebaseUser;
-import com.google.firebase.auth.UserInfo;
-import com.shehabsalah.geranyapp.R;
-import com.firebase.ui.auth.AuthUI;
-import com.shehabsalah.geranyapp.util.Config;
-
-import java.util.Arrays;
-
-/**
- * Created by ShehabSalah on 7/26/17.
- * This class contain the main configurations of the parent of Activities
- */
-
-public class ApplicationMain extends AppCompatActivity {
-    private final String TAG = ApplicationMain.class.getSimpleName();
-    protected static int RC_SIGN_IN = 0;
-    protected FirebaseAuth mAuth;
-    protected FirebaseAuth.AuthStateListener mAuthListener;
-    protected UserInfo userInfo;
-
-
-    /**
-     * This method check if the user exists or not
-     * @return true if the user exists and false if the user not exist
-     * */
-    protected boolean isNotLoggedIn() {
-        FirebaseUser user = FirebaseAuth.getInstance().getCurrentUser();
-        return user==null;
-    }
-    /**
-     * This method responsible on creating the login layout of the Login page.
-     * Login page support Facebook & google authentications only.
-     * */
-    protected void signIn(){
-        startActivityForResult(
-                AuthUI.getInstance()
-                        .createSignInIntentBuilder()
-                        .setProviders(Arrays.asList(new AuthUI.IdpConfig.Builder(AuthUI.GOOGLE_PROVIDER).build(),
-                                new AuthUI.IdpConfig.Builder(AuthUI.FACEBOOK_PROVIDER).build()))
-                        .setLogo(R.mipmap.gerany_logo_png)
-                        .setTheme(R.style.LoginStyle)
-                        .setIsSmartLockEnabled(false)
-                        .build(),
-                RC_SIGN_IN);
-    }
-
-    /**
-     * This method is connecting tha application with Firebase Authentication
-     * */
-    protected void setFirebaseAuthListener(){
-        mAuthListener = new FirebaseAuth.AuthStateListener() {
-            @Override
-            public void onAuthStateChanged(@NonNull FirebaseAuth firebaseAuth) {
-                FirebaseUser user = firebaseAuth.getCurrentUser();
-                if (user != null) {
-                    // Even a user's provider-specific profile information
-                    // only reveals basic information
-                        UserInfo profile = user.getProviderData().get(0);
-                        // Id of the provider (ex: google.com)
-                        String providerId = profile.getProviderId();
-                        // UID specific to the provider
-                        String profileUid = profile.getUid();
-                        // Name, email address, and profile photo Url
-                        String profileDisplayName = profile.getDisplayName();
-                        String profileEmail = profile.getEmail();
-                        Uri profilePhotoUrl = profile.getPhotoUrl();
-                        Log.v("LOGIN",providerId);
-                        Log.v("LOGIN",profileUid);
-                        Log.v("LOGIN",profileDisplayName);
-                        if (profileEmail!=null)
-                            Log.v("LOGIN",profileEmail);
-                        Log.v("LOGIN",profilePhotoUrl.toString());
-                        userInfo = profile;
-
-                    Log.d("LOGIN", "onAuthStateChanged:signed_in:" + user.getUid());
-                } else {
-                    // User is signed out
-                    Log.d("mesh shebo", "onAuthStateChanged:signed_out");
-                }
-                // ...
-            }
-        };
-    }
-
-
-    /**
-     * getUserInfo is getting the current connecting user information
-     * @return UserInfo Object which contain the basic information of the user
-     * */
-    protected UserInfo getUserInfo(){
-        FirebaseUser user = FirebaseAuth.getInstance().getCurrentUser();
-        return user!=null?user.getProviderData().get(0):null;
-    }
-
-    /**
-     * This method responsible on handling the sign in states such as: sign in cancellation,
-     * no network to sign in and unknown error happen. Each state will produce different message
-     * to the user.
-     * @param response The sign in response to identify the state
-     * @param resultCode The onActivityResult passed code to make sure that sign in doesn't succeeded
-     * */
-    protected void handleSignInStates(int resultCode, IdpResponse response){
-        if (resultCode == ResultCodes.OK) {
-            FirebaseUser user = FirebaseAuth.getInstance().getCurrentUser();
-            if (user != null) {
-                // Do Nothing
-            } else {
-                // User is signed out of Firebase
-            }
-        } else {
-            // Sign in failed
-            if (response == null) {
-                // User pressed back button
-                Log.v("LOGIN","SIGNIN_CANCELED");
-                finish();
-                return;
-            }
-
-            if (response.getErrorCode() == ErrorCodes.NO_NETWORK) {
-                Log.v("LOGIN","NO_NETWORK");
-                Config.toastLong(this, getString(R.string.no_internet));
-                return;
-            }
-
-            if (response.getErrorCode() == ErrorCodes.UNKNOWN_ERROR) {
-                Log.v("LOGIN","UNKNOWN_ERROR");
-                Config.toastLong(this, getString(R.string.unknown_error));
-            }
-        }
-
-    }
-
-}
Index: app/src/main/java/com/shehabsalah/geranyapp/util/Config.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/com/shehabsalah/geranyapp/util/Config.java	(revision ab15501144c4075fcc31f40c938c983632c6f1c6)
+++ app/src/main/java/com/shehabsalah/geranyapp/util/Config.java	(revision )
@@ -12,7 +12,8 @@
 public class Config {
 
     //Fragments
-    public static final String ADD_NEW_PLACE_FRAGMENT = "add_new_place_fragment";
+    public static final String ADD_NEW_PLACE_FRAGMENT = "addNewPlaceFragment";
+    public static final String MY_NEW_PLACES_LIST = "myNewPlacesList";
 
 
     /**
Index: app/src/main/res/layout/activity_main.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/res/layout/activity_main.xml	(revision ab15501144c4075fcc31f40c938c983632c6f1c6)
+++ app/src/main/res/layout/activity_main.xml	(revision )
@@ -5,36 +5,16 @@
     android:layout_width="match_parent"
     android:layout_height="match_parent"
     android:background="@color/appBackground"
-    tools:context="com.shehabsalah.geranyapp.activities.MainActivity">
+    tools:context=".views.activities.MainActivity">
     <FrameLayout
         android:id="@+id/home_container"
         android:layout_width="match_parent"
         android:layout_height="match_parent"
-        android:layout_marginBottom="0dp"
         app:layout_constraintBottom_toTopOf="@+id/navigation"
         app:layout_constraintTop_toTopOf="parent"
-        android:layout_marginTop="0dp"
-        android:layout_marginLeft="0dp"
         app:layout_constraintLeft_toLeftOf="parent"
-        android:layout_marginRight="0dp"
         app:layout_constraintRight_toRightOf="parent"/>
 
-
-    <TextView
-        android:id="@+id/message"
-        android:layout_width="0dp"
-        android:layout_height="wrap_content"
-        android:text="@string/title_home"
-        android:textAlignment="center"
-        android:layout_marginRight="8dp"
-        app:layout_constraintRight_toRightOf="parent"
-        android:layout_marginLeft="8dp"
-        app:layout_constraintLeft_toLeftOf="parent"
-        android:layout_marginBottom="8dp"
-        app:layout_constraintBottom_toTopOf="@+id/sign_out"
-        android:layout_marginStart="8dp"
-        android:layout_marginEnd="8dp"
-        android:visibility="gone"/>
     <Button
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
@@ -43,16 +23,10 @@
         android:onClick="onClick"
         android:background="@color/colorAccent"
         android:textColor="#fff"
-        android:layout_marginRight="8dp"
         app:layout_constraintRight_toRightOf="parent"
-        android:layout_marginLeft="8dp"
         app:layout_constraintLeft_toLeftOf="parent"
         app:layout_constraintBottom_toBottomOf="parent"
-        android:layout_marginBottom="8dp"
         app:layout_constraintTop_toTopOf="parent"
-        android:layout_marginTop="8dp"
-        android:layout_marginStart="8dp"
-        android:layout_marginEnd="8dp"
         android:visibility="gone"/>
     <android.support.design.widget.BottomNavigationView
         android:id="@+id/navigation"
Index: app/src/main/res/values/strings.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/res/values/strings.xml	(revision ab15501144c4075fcc31f40c938c983632c6f1c6)
+++ app/src/main/res/values/strings.xml	(revision )
@@ -13,5 +13,20 @@
     <string name="add_new_place_text">Do you like to add your current place to your places list?</string>
     <string name="no_signal_message">There is no internet connection :(</string>
     <string name="current_place_exist">Your current place in your places list</string>
+    <string name="name_your_place">Do you like to give a name to your place? You can change it later.</string>
+    <string name="place_added">Your place has been added successfully!</string>
     <string name="add_new_place_button">Add To My Places</string>
+
+    <string name="animate_add_location">add_location_animation</string>
+
+
+    <!--Edit Location Dialog-->
+    <string name="edit_location_dialog_header">Edit Places</string>
+    <string name="edit_location_dialog_description">Rename the current place</string>
+    <string name="save_button">SAVE</string>
+    <string name="cancel_button">CANCEL</string>
+    <string name="location_dialog_max_length">0 / 50</string>
+    <string name="formatting_max_length" formatted="false">%d / 50</string>
+    <string name="empty_name_error_message">"Cannot save an empty name! Please enter the place name."</string>
+
 </resources>
Index: app/src/main/AndroidManifest.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/AndroidManifest.xml	(revision ab15501144c4075fcc31f40c938c983632c6f1c6)
+++ app/src/main/AndroidManifest.xml	(revision )
@@ -15,7 +15,7 @@
         android:supportsRtl="true"
         android:theme="@style/AppTheme"
         tools:replace="android:supportsRtl">
-        <activity android:name=".activities.MainActivity"
+        <activity android:name=".views.activities.MainActivity"
             android:screenOrientation="portrait">
             <intent-filter>
                 <action android:name="android.intent.action.MAIN" />
\ No newline at end of file
